### PR TITLE
feat: added zkEVM bridge between L1 and L2

### PR DESCRIPTION
**Pull request by impelcrypto**
**Pull Request Summary**

* feat: added zkEVM bridge between L1 and L2
* feat: updated Assets page for token transfer
* feat: update endpoints from Akiba to zKatana

Info:
* The portal (production page) will be updated on 23-Oct 

**Check list**

- [ ] contains breaking changes
- [x] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Adds**
* feat: added zkEVM bridge between L1 and L2

image

image

image

* feat: updated Assets and token transfer page for zkEVM networks
image

image

image

### CODE REVIEW

File: src/components/bridge/ethereum/L1History.vue [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    const { t } = useI18n();
+
+    const syncIndex = ref<number | undefined>();
+
+    const handleUpdateHistory = async (index: number): Promise<void> => {
+      try {
+        syncIndex.value = index;
+        await props.fetchUserHistory();
+      } catch (error) {
+        console.error(error);
```
Comment (bobo-k2) [inline src/components/bridge/ethereum/L1History.vue:200]: Does this error needs to be show to an user?

**Review by bobo-k2 — CHANGES_REQUESTED** [review]
Code LGTM. General comment is that there are lot of `console.log` inside `catch`. Consider if some of those errors should be displayed to an user. If not, than fine

I also se a lot of errors in the console. They seems like RPC endpoint related
image

**Comment by impelcrypto** [comment]
> There is one error which might require attention

Ok, will take a look in this or different PR

**Comment by impelcrypto** [comment]
> I also se a lot of errors in the console. They seems like RPC endpoint related

hmm... we might have to ask Gelato about the CORS issue.

**Comment by Kahonnohak** [comment]
Bridge and transfer works perfectly Great work!

- Lets remove the Bridge from sidebar for this moment and will bring this back when all polish is done before the mainnet
- As discussed, disable the Dashabord for zKatana please
Screenshot 2023-10-17 at 10 13 14

- Please link to Alchemy faucet
Screenshot 2023-10-17 at 10 13 30

- Please use this icon as zKatana network icon for header and all chain select

Astar_zkEVM_network_icon

**Comment by impelcrypto** [comment]
> I also se a lot of errors in the console. They seems like RPC endpoint related

> There is one error which might require attention

@bobo-k2 
Could you share with me how to reproduce the CORS error? It's fine on my end.

image

File: src/boot/api.ts [hunk 2]
```diff
@@ -115,18 +115,29 @@ export default boot(async ({ store }) => {
     }
   });
 
+  const setWeb3 = async (networkId: TNetworkId): Promise<void> => {
+    const web3 = await createAstarWeb3Instance(networkId);
+    if (!web3) {
+      console.error(`cannot create the web3 instance with network id ${networkId}`);
```
[the remark this answers is not in the document]
Comment (impelcrypto) [reply src/boot/api.ts:121]: This is ok, actually I have never seen the error here

File: src/components/bridge/ethereum/L1History.vue (hunk 1 above)
Comment (impelcrypto) [reply src/components/bridge/ethereum/L1History.vue:200]: I think no need

**Comment by impelcrypto** [comment]
@bobo-k2 

> General comment is that there are lot of console.log inside catch. Consider if some of those errors should be displayed to an user. If not, than fine

I believe there are fine. I have a rule that put a `console.error(error)` in the every `catch` scope no matter any case. This will help developer to see the error which is occurred on `try` scope and the error shouldn't happen on production. I'll display the error message if it is useful for users. 

FYI: Usually, I won't push code with `console.log`. I use `console.info` or `console.error` if I want to display something on the console.

File: src/boot/api.ts [hunk 3]
```diff
@@ -115,18 +115,29 @@ export default boot(async ({ store }) => {
...
   watchPostEffect(async () => {
     store.commit('general/setChainInfo', chainInfo.value);
-    if (chainInfo.value?.chain) {
-      const currentChain = chainInfo.value?.chain as ASTAR_CHAIN;
-      const currentNetworkIdx = getProviderIndex(currentChain);
-      const web3 = await createAstarWeb3Instance(currentNetworkIdx as TNetworkId);
-      if (!web3) {
-        console.error(`cannot create the web3 instance with network id ${currentNetworkIdx}`);
+    const networkIdx = store.getters['general/networkIdx'];
+    const isZkEvm = networkIdx === endpointKey.ZKATANA || networkIdx === endpointKey.ASTAR_ZKEVM;
```
[the remark this answers is not in the document]
Comment (impelcrypto) [reply src/boot/api.ts]: Fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/960/commits/8293677694de686b99aca08ab4b6027885f2b8da) commit.

**Comment by bobo-k2** [comment]
> @bobo-k2 Could you share with me how to reproduce the CORS error? It's fine on my end.

I didn't do anything special. Just opened the portal and bridged tokens. Not when exactly the error happened. Let me try again

**Comment by impelcrypto** [comment]
@Kahonnohak 
I've updated zKatana logo

image

**Comment by Kahonnohak** [comment]
Please update FAQ - remove all that we have on zkEVM network at this moment and add below please.

What is Astar zkEVM?
https://docs.astar.network/docs/build/zkEVM/

How to get test gas tokens (sETH)?
https://docs.astar.network/docs/build/zkEVM/zkevm-faucet

How to bridge tokens to Astar zkEVM?
https://docs.astar.network/docs/build/zkEVM/bridge-to-zkevm

**Comment by impelcrypto** [comment]
@Kahonnohak Updated FAQs.

image

**Review by bobo-k2 — APPROVED** [review]
Great work. LGTM

**Review by Kahonnohak — APPROVED** [review]
Great work @impelcrypto, thanks for the great support @bobo-k2

**Review by gluneau — APPROVED** [review]
Great work!

**Review by ayumitk — APPROVED** [review]
